### PR TITLE
Update how WooCommerce Nonce compatibility works

### PR DESCRIPTION
**Pull request by ictbeheer**
This is a regression fix for #894 which fixes the problem that WooCommerce causes by manipulating the nonce for logged-out users.

#2337

### CODE REVIEW

File: includes/functions.php [hunk 1]
```diff
@@ -64,7 +64,7 @@ function caldera_forms_fix_license_manger_link(){
  */
 function caldera_forms_woo_nonce_fix( $user_id, $action) {
 	if ( class_exists( 'WooCommerce' ) ) {
-		if ( $user_id && $user_id != 0 && $action && $action == 'caldera_forms_front' ) {
+		if ( $user_id && $user_id !== 0 && $action && strpos( $action, 'caldera_forms_front' ) !== false ) {
```
Comment (Shelob9) [inline includes/functions.php]: @ictbeheer We use yoda conditionals. The old code is wrong in this manner.

`if ( $user_id &&  0 != $user_id && $action && 0 === strpos( $action, 'caldera_forms_front' )  )` would be more appropriate. Also, using `0 === strpos(`... ensures it is the start of the string `$action`

**Comment by Shelob9** [comment]
@ictbeheer Thanks for your pull request. I left a comment on code style, but agree with your assessment of the bug/fix. If you can please update your PR, I will merge and include in 1.6 release.

**Comment by ictbeheer** [comment]
@Shelob9 thanks for your review, I have updated the PR. I do have an additional question though. As Caldera Forms uses the WP Rest API should we also check for the `wp_rest` nonce?

**Comment by Shelob9** [comment]
@ictbeheer Thanbks for the update, I will merge now.

>As Caldera Forms uses the WP Rest API should we also check for the wp_rest nonce?
I am very uncomfortable changing how that nonce works. I'll listen to a strong case for why I should, but the risk of making that nonce behave not as anyone would assume it does is scary to me.

Also, WooCommerce should play nice with that nonce, it's in WordPress core. Caldera Forms shouldn't have to work around WooCommerce like this, but whatever they are bigger and it's easier for us to work around them -- though it leads to having to do things like this.

**Pull request merged**